src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -239,9 +240,14 @@ def get_keyring_with_caps(self, entity: AuthEntity, caps: List[str]) -> str:
                 self.mgr.log.warning(f"Unable to update caps for {entity}")
         return keyring
 
-    def _inventory_get_addr(self, hostname: str) -> str:
-        """Get a host's address with its hostname."""
-        return self.mgr.inventory.get_addr(hostname)
+    def _inventory_get_fqdn(self, hostname: str) -> str:
```

Comment:
http://pulpito.front.sepia.ceph.com/adking-2022-03-18_02:57:32-orch:cephadm-wip-adk-testing-2022-03-17-1930-distro-basic-smithi/

```
Failed to apply grafana spec GrafanaSpec.from_json(yaml.safe_load('''service_type: grafana
  service_name: grafana
  placement:
    count: 1
  ''')): 'GrafanaService' object has no attribute '_inventory_get_addr'
  Traceback (most recent call last):
    File "/usr/share/ceph/mgr/cephadm/serve.py", line 503, in _apply_all_services
      if self._apply_service(spec):
    File "/usr/share/ceph/mgr/cephadm/serve.py", line 769, in _apply_service
      daemon_spec = svc.prepare_create(daemon_spec)
    File "/usr/share/ceph/mgr/cephadm/services/monitoring.py", line 24, in prepare_create
     daemon_spec.final_config, daemon_spec.deps = self.generate_config(daemon_spec)
    File "/usr/share/ceph/mgr/cephadm/services/monitoring.py", line 45, in generate_config
      addr = daemons[0].ip if daemons[0].ip else self._inventory_get_addr(daemons[0].hostname)
  AttributeError: 'GrafanaService' object has no attribute '_inventory_get_addr'
```

This needs to be rebased after https://github.com/ceph/ceph/pull/44751 merged as that added another call to `_inventory_get_addr`